src/data.rs

```diff
@@ -105,6 +110,37 @@ mod array_impls {
         })))
     }
 
+    pub(crate) fn create_boolean_array_with_errors<DT: CellType + DataType + Debug>(
+        data: &Range<DT>,
+        col: usize,
+        offset: usize,
+        limit: usize,
+    ) -> (Arc<dyn Array>, CellErrors) {
+        let mut cell_errors = CellErrors { errors: vec![] };
+
+        let arr = Arc::new(BooleanArray::from_iter((offset..limit).map(|row| {
+            data.get((row, col)).and_then(|cell| {
+                if cell.is_empty() {
+                    None
+                } else if let Some(b) = cell.get_bool() {
+                    Some(b)
+                } else if let Some(i) = cell.get_int() {
+                    Some(i != 0)
+                } else if let Some(f) = cell.get_float() {
+                    Some(f != 0.0)
+                } else {
+                    cell_errors.errors.push(CellError {
+                        position: (row - offset, col),
+                        detail: format!("Expected boolean but got '{:?}", cell),
+                    });
+                    None
+                }
+            })
+        })));
+
+        (arr, cell_errors)
```

Comment:
```suggestion
        (arr, CellErrors { errors })
```

Reply:
Did it the way suggested in your other comment